src/hotspot/os/bsd/os_bsd.cpp

```diff
@@ -2208,11 +2208,11 @@ static inline struct timespec get_mtime(const char* filename) {
 int os::compare_file_modified_times(const char* file1, const char* file2) {
   struct timespec filetime1 = get_mtime(file1);
   struct timespec filetime2 = get_mtime(file2);
-  int diff = filetime1.tv_sec - filetime2.tv_sec;
+  time_t diff = filetime1.tv_sec - filetime2.tv_sec;
```

Comment:
Is time_t guaranteed to be signed?
```suggestion
  int diff = primitive_compare(filetime1.tv_sec, filetime2.tv_sec);
```